client/lib/resize-image-url/index.js

```diff
@@ -61,32 +61,35 @@ const scaleByFactor = value => value * IMAGE_SCALE_FACTOR;
  * URL containing query string arguments is passed to this function, it will
  * return `null`.
  *
- * @param   {String}          imageUrl Original image url
- * @param   {(Number|Object)} resize   Resize pixel width, or object of query
+ * @param   {string}          imageUrl Original image url
+ * @param   {(number|object)} resize   Resize pixel width, or object of query
  *                                     arguments (assuming Photon or Gravatar)
- * @param   {?Number}         height   Pixel height if specifying resize width
- * @param   {?Boolean}        makeSafe Should we make sure this is on a safe host?
- * @returns {?String}                  Resized image URL, or `null` if unable to resize
+ * @param   {?number}         height   Pixel height if specifying resize width
+ * @param   {?boolean}        makeSafe Should we make sure this is on a safe host?
+ * @returns {?string}                  Resized image URL, or `null` if unable to resize
  */
 export default function resizeImageUrl( imageUrl, resize, height, makeSafe = true ) {
 	if ( 'string' !== typeof imageUrl ) {
 		return imageUrl;
 	}
 
-	const parsedUrl = parse( imageUrl, true, true );
-	if ( ! REGEXP_VALID_PROTOCOL.test( parsedUrl.protocol ) ) {
+	const urlType = determineUrlType( imageUrl );
+
+	if ( urlType !== URL_TYPE.ABSOLUTE ) {
+		// Invalid URL; return unmodified.
 		return imageUrl;
 	}
-	if ( ! parsedUrl.hostname ) {
-		// no hostname? must be a bad url.
+
+	const parsedUrl = new URL( imageUrl, 'https://__domain__.invalid' );
+
+	if ( ! REGEXP_VALID_PROTOCOL.test( parsedUrl.protocol ) ) {
 		return imageUrl;
 	}
 
-	parsedUrl.query = omit( parsedUrl.query, SIZE_PARAMS );
+	SIZE_PARAMS.forEach( param => parsedUrl.searchParams.delete( param ) );
```

Comment:
noting that in addition to removing `lodash` here we are also introducing mutation into the variable which is referenced through multiple lines of this function. this may be fine, but it's something we like to avoid in general so that we don't have to interpret the code as much in our heads when reading, don't have to pay attention to what might have changed it.

faster than the original, slower than your change…

```js
const withoutSizes = SIZE_PARAMS.reduce( ( finalUrl, nextSize ) => {
	finalUrl.searchParams.delete( nextSize );
	return finalUrl;
}, new URL( url.toString() );
```

This is only marginally different but by cloning the URL first we can avoid that mutation. Could also keep the `forEach` and clone separately if on purity grounds we don't like mutating inside a `reduce`

```js
const withoutSizes = new URL( url.toSring() );
SIZE_PARAMS.forEach( param => withoutSizes.searchParams.delete( param ) );
```

This way although the mutation is there it's isolated and colocated with the initial assignment.

Reply:
I don't really see the issue with mutating an object in which you're preparing a result. Is it the fact that it's named `parsedUrl`? I'll change the name to `resultUrl` to make the intention clearer.